src/SchedulerSubsystem/Scheduler.java

```diff
@@ -1,40 +1,49 @@
 package SchedulerSubsystem;
 
-import Networking.Direction;
-import Networking.Messages.DestinationEvent;
-import Networking.Messages.ElevatorStateEvent;
-import Networking.Messages.MoveElevatorCommand;
-import Networking.Messages.SystemMessage;
-import Networking.Receivers.DMA_Receiver;
-import Networking.Transmitters.DMA_Transmitter;
+import Messaging.Commands.MoveElevatorCommand;
+import Messaging.Direction;
+import Messaging.Events.DestinationEvent;
+import Messaging.Events.ElevatorStateEvent;
+import Messaging.Events.FloorRequestEvent;
+import Messaging.Receivers.DMA_Receiver;
+import Messaging.SystemMessage;
+import Messaging.Transmitters.DMA_Transmitter;
 import com.sun.jdi.InvalidTypeException;
-import org.junit.jupiter.api.*;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.runner.RunWith;
 
-import java.util.*;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
 
-import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
+@RunWith(Enclosed.class)
 public class Scheduler implements Runnable {
     private final DMA_Transmitter transmitterToFloor;
     private final DMA_Transmitter transmitterToElevator;
     private final DMA_Receiver receiver;
-    private Set<DestinationEvent> floorRequests;
+    private final Map<DestinationEvent, Long> floorRequestsToTime;
 
     public Scheduler(DMA_Receiver receiver, DMA_Transmitter transmitterToFloor, DMA_Transmitter transmitterToElevator) {
         this.receiver = receiver;
         this.transmitterToElevator = transmitterToElevator;
         this.transmitterToFloor = transmitterToFloor;
-        floorRequests = new HashSet<DestinationEvent>();
+        floorRequestsToTime = new HashMap<>();
     }
 
     /**
      * Process event received from the floor.
      *
-     * @param destinationEvent Destination event received from floor.
+     * @param event Event to process.
      */
-    private void processDestinationEvent(DestinationEvent destinationEvent) {
+    private void storeFloorRequest(FloorRequestEvent event) {
         // Store event locally to use in scheduling
-        floorRequests.add(destinationEvent);
+        floorRequestsToTime.put(event.destinationEvent(), event.time());
```

Comment:
We would like to keep the OLDEST request, meaning we should not update it if the key already exists otherwise we could starve the floor.